trade-finance-manager-api/src/v1/controllers/number-generator.controller.js

```diff
@@ -45,63 +48,83 @@ const checkAzureNumberGeneratorFunction = async () => {
+      type: { $eq: CONSTANTS.DURABLE_FUNCTIONS.TYPE.NUMBER_GENERATOR },
+    })
+    .toArray();
+  try {
+    const taskResults = runningTasks.map(({ numberGeneratorFunctionUrls = {} }) =>
+      api.getFunctionsAPI(
+        CONSTANTS.DURABLE_FUNCTIONS.TYPE.NUMBER_GENERATOR,
+        numberGeneratorFunctionUrls.statusQueryGetUri
+      ),);
+
+    const taskResultsList = await Promise.all(taskResults);
+
+    taskResultsList.forEach(async (task) => {
+      if (task?.runtimeStatus === 'Completed' && !task?.output?.ukefId?.error) {
+        // Only process if all tasks for that deals have finished
+        if (otherDealTasksStillRunning(task, taskResultsList)) {
           return;
+        }
+
+        const { input, output, instanceId, runtimeStatus } = task;
+
+        // Update portalDeal
+        switch (input.dealType) {
+          case CONSTANTS.DEALS.DEAL_TYPE.BSS_EWCS:
+            await updatePortalDeal(input, output);
+            break;
+
+          case CONSTANTS.DEALS.DEAL_TYPE.GEF:
+            await updateGefApplication(input, output);
+            break;
+
+          default:
+            return;
+        }
+
+        // Submit to TFM.
+        // Only trigger this update if the Azure function task's ID, is the same as the deal ID.
+        // Without this conditional, every task (i.e multiple facilities) will trigger multiple deal submissions.
+        if (input.entityId === input.dealId) {
+          await dealSubmitController.submitDealAfterUkefIds(input.entityId, input.dealType, input.user);
+        }
+
+        // Update functionLog
+        // Keep any with errors for reference but remove successful ones
+        if (output && output.error) {
+          await collection.findOneAndUpdate(
+            { instanceId: { $eq: instanceId } },
+            $.flatten({
+              status: runtimeStatus,
+              taskResult: task,
+            }),
+          );
+        } else if (typeof instanceId === 'string') {
+          await collection.deleteOne({ instanceId: { $eq: instanceId } });
+        }
       }
-
-      // Submit to TFM.
-      // Only trigger this update if the Azure function task's ID, is the same as the deal ID.
-      // Without this conditional, every task (i.e multiple facilities) will trigger multiple deal submissions.
-      if (input.entityId === input.dealId) {
-        await dealSubmitController.submitDealAfterUkefIds(input.entityId, input.dealType, input.user);
-      }
-
-      // Update functionLog
-      // Keep any with errors for reference but remove successful ones
-      if (output && output.error) {
-        await collection.findOneAndUpdate(
-          { instanceId: { $eq: instanceId } },
-          $.flatten({
-            status: runtimeStatus,
-            taskResult: task,
-          }),
-        );
-      } else if (typeof instanceId === 'string') {
-        await collection.deleteOne({ instanceId: { $eq: instanceId } });
-      }
-    }
-  });
+    });
+  } finally {
```

Comment:
My main concern is if any error is thrown I want the db to be reset so it can try again.
Better solution might be putting this in 106, and having a check for how long ago the last attempt to update was?

something like
```
if (new Date() - document.startedAt < 1 day) {
 // Discard document from list
}
```